src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -1586,7 +1586,22 @@ private int GetDataIntoOleStructsByTypeMedimIStream(string format, object data,
 
                         byte[] buffer = new byte[NativeMethods.IntPtrToInt32(size)];
                         inkStream.Position = 0;
-                        inkStream.Read(buffer, 0, NativeMethods.IntPtrToInt32(size));
+#if NET7_0_OR_GREATER
+                inkStream.ReadExactly(buffer, 0, NativeMethods.IntPtrToInt32(size));
```

Comment:
What kind of stream can inkStream be? If it can be an arbitrary stream, are there tests (or can we add tests) validating what happens when the stream provides back fewer bytes than requested?

Reply:
I typically do this by writing a type that derives from MemoryStream and overrides the Read methods to return back a maximum of one byte, e.g.
```C#
internal sealed class TrickleStream : MemoryStream
{
    public override int Read(byte[] buffer, int offset, int count) =>
        base.Read(buffer, offset, Math.Min(count, 1));
    ...
}
```

Reply:
Oh this particular one is actually from OLE, so in most cases it might be limited to IStream, although within application you can still store any Stream in the DataStore. This particular code doesn't have many checks around, unlike the deserialization. If Read returns 0, this just writes 0 to the medium and calls it a day. I can see how if we start throwing here, applications that did not use to crash can start crashing. Still probably a reasonably thing to do.